src/settings/MappingProfiles/detailsSections/edit/OrderDetailSection/CostDetails.js

```diff
@@ -106,16 +107,19 @@ export const CostDetails = ({
           />
         </Col>
         <Col xs={4}>
-          <Field
-            component={Checkbox}
+          <Checkbox
             label={<FormattedMessage id={`${TRANSLATION_ID_PREFIX}.order.costDetails.field.useSetExchangeRate`} />}
-            name={getBoolFieldName(54)}
-            onChange={e => {
-              setIsSetExchangeDisabled(!(e.target.value === BOOLEAN_ACTIONS.ALL_FALSE));
+            onChange={() => {
+              setIsUseExchangeChecked(prevState => !prevState);
+              setIsSetExchangeDisabled(prevState => {
+                if (!prevState) {
+                  setReferenceTables(getFieldName(51), '');
+                }
+
+                return !prevState;
+              });
```

Comment:
I would create a separate function outside `return` block to get rid of possible unnecessary rerenders.